lib/utils.js

```diff
@@ -302,55 +305,88 @@ function prettifyObject ({
   skipKeys = [],
   customPrettifiers = {},
   errorLikeKeys = ERROR_LIKE_KEYS,
-  excludeLoggerKeys = true
+  excludeLoggerKeys = true,
```

Comment:
This function is much longer with this PR. Are we able to factor some of the code into smaller functions so that it is easier to reason through?

Reply:
Yeah I plan to, just wanted to make sure the overall design was good before I clean up the code.

Reply:
I have limited time to review these PRs. Please put forth the easiest to follow version of it. In this case, some named functions that describe the actions being performed would be very helpful.

Reply:
I just cleaned up the code a bit. I only split out one function, `prettifyErrors`, because a lot of the logic is inter-dependent so it's hard to split it across function boundaries. I did de-duplicate some shared logic as well though which makes the function body shorter, and also added some comments. let me know if that's easier to follow.

Reply:
Yes, that helped a lot.